handlers/http_test.go

```diff
@@ -226,8 +226,68 @@ func TestHandleWhenErrWithoutStatus(t *testing.T) {
 	limiter.ServeHTTP(w, r)
 	assertNoRateLimitHeaders(t, w.Header())
 
-	if w.Code != 500 {
-		t.Fatalf("expected status 500, received %d", w.Code)
+	if w.Code != http.StatusInternalServerError {
+		t.Fatalf("expected status %d, received %d", http.StatusInternalServerError, w.Code)
 	}
 	limitMock.AssertExpectations(t)
 }
+
+// Test cases when an error occurs and either value of AllowOnError
+var allowOnErrorCases = []struct {
+	AllowOnError      bool
+	ExpectedCode      int
+	RateLimiterStatus []ratelimiter.Status
+}{
+	// It should still block if AllowOnError == false
+	{
+		false,
+		http.StatusInternalServerError,
+		[]ratelimiter.Status{},
+	},
+	// If AllowOnError == true and
+	{
+		true,
+		http.StatusOK,
+		[]ratelimiter.Status{},
+	},
+	// It should still add headers if AllowOnError == true
+	{
+		true,
+		http.StatusOK,
+		[]ratelimiter.Status{sphinxStatus},
+	},
```

Comment:
Because we've changed the fnality to not add headers w/ AllowOnError + Error, the tests have materially changed. 

1871bc6